src/main/kotlin/compiler/ast/N_LITERAL.kt

```diff
@@ -54,3 +56,27 @@ class N_LITERAL_MAP(val value: Map<N_EXPR, N_EXPR>): N_LITERAL() {
         coder.value(this, value.size)
     }
 }
+
+class N_LITERAL_FUN(val args: List<N_IDENTIFIER>, val block: N_BLOCK): N_LITERAL() {
+    override fun kids() = args + listOf(block)
+    override fun code(coder: Coder) {
```

Comment:
To compile the function declaration, we put the argnames, captured varnames, and entry point on the stack, then code O_FUNVAL which will turn this into a VFun on the stack.  We then jump past the compiled function block; it will be executed into by the VFun invocation.